tls/s2n_handshake_io.c

```diff
@@ -694,7 +694,16 @@ static int s2n_advance_message(struct s2n_connection *conn)
 }
 
 int s2n_generate_new_client_session_id(struct s2n_connection *conn)
+    CONTRACT_REQUIRES(__CPROVER_is_fresh(conn, sizeof(*conn)) || (conn == NULL))
+    CONTRACT_ASSIGNS_ERR(conn->session_id_len, conn->session_id)
+    CONTRACT_ENSURES(S2N_IMPLIES(conn == NULL, __CPROVER_return_value == S2N_FAILURE))
+    CONTRACT_ENSURES(S2N_IMPLIES(__CPROVER_return_value == S2N_SUCCESS && conn->mode == S2N_SERVER, conn->session_id_len == S2N_TLS_SESSION_ID_MAX_LEN))
```

Comment:
Should we replace the `return 0` at line 715 with `return S2N_SUCCESS` for clarity?

Reply:
YES!!!!

Reply:
Agree